s2nTests/scripts/awslc-entrypoint.sh

```diff
@@ -13,6 +13,8 @@ cp /saw-bin/z3-4.8.8 bin/z3
 
 export PATH=/saw-script/aws-lc-verification/SAW/bin:$PATH
 export CRYPTOLPATH=/saw-script/aws-lc-verification/cryptol-specs
+export SAW_SOLVER_CACHE_PATH=/saw-cache
+saw --clean-solver-cache
```

Comment:
Having looked at this `clean_solver_cache` command with fresh eyes, I wonder if the name is misleading. I was initially confused because I thought that this script was completely cleaning out the cache before running the `awslc` proofs, which would be completely counterproductive. But that isn't what actually happens: it only cleans the _out-of-date_ entries in the cache, which is much more sensible that cleaning _all_ of the cache.

I wonder if renaming this option to something like `--clean-stale-solver-cache` or `--clean-outdated-solver-cache` would make this more obvious. This would induce some amount of code churn, but I think this name would more clearly reflect its intended purpose.

Reply:
Ah, excellent point! I renamed this argument and the SAW command to `clean_mismatched_versions_solver_cache` in a6600cd651636c243cc084e3de78ec61a0003001. This is a bit wordier, but makes it clear exactly what's happening. I'm worried something like `clean_stale` or `clean_outdated` might give the wrong impression that it's also taking into account the timestamp of when the entry was last used – or that's it's only removing entries with *outdated* version numbers, as opposed to what it's actually doing, which is just removing any entry with a version string which doesn't exactly match what's in the current environment.

Reply:
I like the name you've chosen. It would also be worth renaming the `cleanSolverCache` function to match it.

Reply:
Good catch – renamed `cleanSolverCache` and `cleanCacheOpt` in 498d5d6d9aca7ebf31f83ce663680d581f5d4ed7 to mention mismatched versions